

from django.http import HttpResponseBadRequest

HttpResponseBadRequest() - не ошибка (не унаследован он BaseException)
поэтому просто вызывается без raise
посылает HTTP-response с кододм ответа 400 (bad request)